connect-inject/health_check_resource_test.go

```diff
@@ -0,0 +1,606 @@
+						annotationStatus:  "injected",
+						annotationService: testServiceNameAnnotation,
+					},
+				},
+				Spec: testPodSpec,
+				Status: corev1.PodStatus{
+					HostIP: "127.0.0.1",
+					Phase:  corev1.PodRunning,
+					Conditions: []corev1.PodCondition{{
+						Type:   corev1.PodReady,
+						Status: corev1.ConditionTrue,
+					}},
+				},
+			},
+			nil,
+			"ServiceID \"test-pod-test-service\" does not exist",
+		},
+		{
+			"PodRunning no label - no change",
+			getSupportedTestTypes(testTypesReconcileOnly),
+			false,
+			"",
+			&corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testPodName,
+					Namespace: "default",
+					Annotations: map[string]string{
+						annotationStatus:  "injected",
+						annotationService: testServiceNameAnnotation,
+					},
+				},
+				Spec: testPodSpec,
+				Status: corev1.PodStatus{
+					HostIP: "127.0.0.1",
+					Phase:  corev1.PodRunning,
+					Conditions: []corev1.PodCondition{{
+						Type:   corev1.PodReady,
+						Status: corev1.ConditionTrue,
+					}},
+				},
+			},
+			nil,
+			"",
+		},
+		{
+			"PodRunning no Ready Status - no change",
+			getSupportedTestTypes(testTypesBoth),
+			false,
+			"",
+			&corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testPodName,
+					Namespace: "default",
+					Labels:    map[string]string{labelInject: "true"},
+				},
+				Spec: testPodSpec,
+				Status: corev1.PodStatus{
+					HostIP: "127.0.0.1",
+					Phase:  corev1.PodRunning,
+				},
+			},
+			nil,
+			"",
+		},
+	}
+	for _, work := range []string{testReconcile, testUpsert} {
+		for _, tt := range cases {
+			if _, ok := tt.ValidTests[work]; !ok {
+				continue
+			}
+			t.Run(work+" "+tt.Name, func(t *testing.T) {
+				var err error
+
+				require := require.New(t)
+				// Get a server, client, and handler
+				server, client, resource := testServerAgentResourceAndController(t, tt.Pod)
+				defer server.Stop()
+
+				if tt.InitialState != testDoNotRegister {
+					// Create a passing service
+					server.AddService(t, testServiceNameReg, api.HealthPassing, nil)
+				}
+				if tt.PreCreateHealthCheck {
+					// register the health check if this is not an object create path
+					if tt.InitialState == api.HealthPassing {
+						registerHealthCheck(t, client, tt.InitialState, testCheckNotesPassing)
+					} else {
+						registerHealthCheck(t, client, tt.InitialState, testFailureMessage)
+					}
+				}
+				if work == testUpsert {
+					err = resource.Upsert("", tt.Pod)
+				} else if work == testReconcile {
+					err = resource.Reconcile()
+				}
+				if tt.Err != "" {
+					// used in the cases where we're expecting an error from
+					// the controller/handler, in which case do not check agent
+					// checks as they're relevant/created.
+					require.Error(err, tt.Err)
+					return
+				}
+				require.NoError(err)
+				actual := getConsulAgentChecks(t, client)
+				if tt.Expected == nil || actual == nil {
+					require.Equal(tt.Expected, actual)
+				} else {
```

Comment:
Outside of check for Name and Type, I think every other test case can be covered by `require.Equal(tt.Expected, actual)`

Reply:
The problem is that actual comes back with additional metadata like node-id that is generated on the fly for each test. These are the specific ones we're trying to validate though.